File: 122-best-time-to-buy-and-sell-stock-ii/122-best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = prices.size();
        int prof = 0;
        for(int i=0;i<n-1;i++){
            if(prices[i]<prices[i+1]){
                prof += prices[i+1]-prices[i];
            }
        }
        return prof;
    }
};

File: 122-best-time-to-buy-and-sell-stock-ii/122-best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        ans = 0
        n = len(prices)
        if n<2:
            return 0
        for i in range(1,n):
            ans += max(0,prices[i]-prices[i-1])
        
        return ans